File: services/product-service/infra/src/main/java/io/micro_spring_boot/product_service/infra/repositories/ProductRepository.java
package io.micro_spring_boot.product_service.infra.repositories;

import io.micro_spring_boot.building_blocks.jpa.domain.aggregate.AggregateRepository;
import io.micro_spring_boot.product_service.domain.aggregates.product.Product;

import java.util.List;

public interface ProductRepository extends AggregateRepository<Product> {
    List<Product> findBySkuCodeIn(List<String> skuCodes);
}

File: services/product-service/infra/src/main/kotlin/com/microspringboot/productservice/infra/repositories/ProductRepository.kt
package com.microspringboot.productservice.infra.repositories

import com.microspringboot.building_blocks.jpa.domain.repository.AggregateRepository
import com.microspringboot.productservice.domain.aggregates.products.Product

interface ProductRepository : AggregateRepository<Product> {
    fun findBySkuCodeIn(skuCodes: List<String>): List<Product>
}
